File: keiner_lindarte_aro/instructor_henry/mis_programas/practicas/ruby/practicas_2/calculo_de_tiempo.rb
puts "Que desea convertir:"
puts "1- hora a minutos"
puts "2- minutos a hora"
puts "3- horas a segundos"
puts "4- minutos a segundos"
puts "5- segundos a horas"
puts "6- segundos a minutos"
opcion = gets.chomp.to_i

## digitamos la informacion al usuario para sepa que opciones escoger

## ponemos las variables correspondientes


if  opcion == 1
puts "Cuantas horas desea convertir a minutos"
horas_minutos = gets.chomp.to_f
elsif opcion == 2
puts"Cuantos minutos desea convertir a horas:"
minutos_horas = gets.chomp.to_f
elsif opcion == 3
puts"Cuantos horas desea convertir a segundos:"
horas_segundos = gets.chomp.to_f
elsif opcion == 4
puts"Cuantos minutos desea convertir a segundos:"
minutos_segundos = gets.chomp.to_f
elsif opcion == 5
puts"Cuantos segundos desea convertir a horas:"
segundos_horas = gets.chomp.to_f
elsif opcion == 6
puts"Cuantos segundos desea convertir a minutos:"
segundos_minutos = gets.chomp.to_f
end

## aqui ponemos funciones "def" de cada operacion de tiempo con la varible correspondiente adentro


def horas_a_minutos(horas_minutos)
    (horas_minutos * 60)
end
def minutos_a_horas(minutos_horas)
    (minutos_horas / 60)
end
def horas_a_segundos(horas_segundos)
    (horas_segundos * 3600)
end
def minutos_a_segundos(minutos_segundos)
    (minutos_segundos * 60)
end
def segundos_a_horas(segundos_horas)
    (segundos_horas / 3600)
end
def segundos_a_minutos(segundos_minutos)
    (segundos_minutos / 60)
end

## y mas condicionales para mostras diferentes resultados diferentes.

if  opcion == 1
puts "la conversion de horas a minutos es #{horas_a_minutos(horas_minutos)} minutos"
elsif opcion == 2
puts "la conversion de minutos a horas es #{minutos_a_horas(minutos_horas)} horas"
elsif opcion == 3
puts "la conversion de horas a segundos es #{horas_a_segundos(horas_segundos)} segundos"
elsif opcion == 4
puts "la conversion de minutos a segundos es #{minutos_a_segundos(minutos_segundos)} segundos"
elsif opcion == 5
puts "la conversion de segundos a horas es #{segundos_a_horas(segundos_horas)} horas"
elsif opcion == 6
puts "la conversion de segundos a minutos es #{segundos_a_minutos(segundos_minutos)} minutos"
end

File: keiner_lindarte_aro/instructor_henry/mis_programas/practicas/java/practicas2/calculo_de_tiempo.java
package ADSO2928707.portafolio_aprendices.keiner_lindarte_aro.instructor_henry.mis_programas.practicas.java.practicas2;
import java.util.Scanner;

public class calculo_de_tiempo {

    public static void main(String[] args) {
        
        try (Scanner input = new Scanner (System.in)){

            System.out.println( "Que desea convertir:");
            System.out.println( "1- hora a minutos");
            System.out.println( "2- minutos a hora");
            System.out.println( "3- horas a segundos");
            System.out.println( "4- minutos a segundos");
            System.out.println( "5- segundos a horas");
            System.out.println( "6- segundos a minutos");

            int opcion = 0;
            opcion = input.nextInt();

            double horas_minutos = 0;
            double minutos_horas = 0;
            double horas_segundos = 0;
            double minutos_segundos = 0;
            double segundos_horas = 0;
            double segundos_minutos = 0;


            if  (opcion == 1){
                System.out.println( "Cuantas horas desea convertir a minutos");
                horas_minutos = input.nextDouble();
            }

            else if (opcion == 2){
                System.out.println("Cuantos minutos desea convertir a horas:");
                minutos_horas = input.nextDouble();
            }

            else if (opcion == 3){
                System.out.println("Cuantos horas desea convertir a segundos:");
                horas_segundos = input.nextDouble();
            }

            else if (opcion == 4){
                System.out.println("Cuantos minutos desea convertir a segundos:");
                minutos_segundos = input.nextDouble();
            }

            else if (opcion == 5){
                System.out.println("Cuantos segundos desea convertir a horas:");
                minutos_segundos = input.nextDouble();
            }

            else if (opcion == 6){
                System.out.println("Cuantos segundos desea convertir a minutos:");
                minutos_segundos = input.nextDouble();
            }


            if  (opcion == 1){
                System.out.println( "la conversion de horas a minutos es "+horas_a_minutos(horas_minutos)+" minutos");
                }
            if (opcion == 2){
                System.out.println( "la conversion de minutos a horas es "+minutos_a_horas(minutos_horas)+" horas");
                }
            if (opcion == 3){
                System.out.println( "la conversion de horas a segundos es "+horas_a_segundos(horas_segundos)+" segundos");
                }
            if (opcion == 4){
                System.out.println( "la conversion de minutos a segundos es "+minutos_a_segundos(minutos_segundos)+" segundos");
                }
            if (opcion == 5){
                System.out.println( "la conversion de segundos a horas es "+segundos_a_horas(segundos_horas)+" horas");
                }
            if (opcion == 6){
                System.out.println( "la conversion de segundos a minutos es "+segundos_a_minutos(segundos_minutos)+" minutos");
                }
            
        }
        
        


    }

    private static double horas_a_minutos(double horas_minutos) {

        return horas_minutos * 60;

    }

    private static double minutos_a_horas(double minutos_horas) {

        return minutos_horas / 60;

    }

    private static double horas_a_segundos(double horas_segundos) {

        return horas_segundos * 3600;

    }

    private static double minutos_a_segundos(double minutos_segundos) {

        return minutos_segundos * 60;

    }

    private static double segundos_a_horas(double segundos_horas) {

        return segundos_horas * 60;

    }

    private static double segundos_a_minutos(double segundos_minutos) {

        return segundos_minutos * 60;

    }

}
    

